.github/workflows/modular-arch-quality-gates.yml

```diff
@@ -0,0 +1,479 @@
+---
+# yamllint disable rule:line-length
+name: Modular Architecture - Quality Gates
+
+'on':
+  pull_request:
+    paths:
+      - 'frontend/packages/**'
+      - 'packages/**'  # Future path support
+  workflow_dispatch:
+    inputs:
+      modules:
+        description: 'Space-separated module names to run (optional)'
+        required: false
+        type: string
+
+permissions:
+  contents: read
+  actions: read
+
+env:
+  NODE_VERSION: '20'
+  GO_VERSION: '1.24.3'
+  # Configurable paths for future flexibility
+  PACKAGES_PATHS: 'frontend/packages,packages'
+
+jobs:
+  detect-changes:
+    name: Detect Module Changes
+    runs-on: ubuntu-latest
+    outputs:
+      has-changes: ${{ steps.check-changes.outputs.has-changes }}
+      changed-modules: ${{ steps.check-changes.outputs.changed-modules }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Detect changed modules
+        id: check-changes
+        run: |
+          # yamllint disable rule:line-length
```

Comment:
So I was about to tell you if there wasn't a pre-built action but I would assume we need this just to filter by valid modules right?